meson.build

```diff
@@ -0,0 +1,1852 @@
+if want_microhttpd != 'no'
+  libmicrohttpd = dependency('libmicrohttpd',
+                             version : '>= 0.9.33',
+                             required : want_microhttpd == 'yes')
+  if libmicrohttpd.found()
+    conf.set('HAVE_MICROHTTPD', 1)
+    m4_defines += ['-DHAVE_MICROHTTPD']
+  endif
+else
+  libmicrohttpd = declare_dependency('microhttpd')
+endif
+
+want_libcryptsetup = get_option('libcryptsetup')
+if want_libcryptsetup != 'no'
+  libcryptsetup = dependency('libcryptsetup',
+                             version : '>= 1.6.0',
+                             required : want_libcryptsetup == 'yes')
+  if libcryptsetup.found()
+    conf.set('HAVE_LIBCRYPTSETUP', 1)
+  endif
+else
+  libcryptsetup = declare_dependency('libcryptsetup')
+endif
+
+want_libcurl = get_option('libcurl')
+if want_libcurl != 'no'
+  libcurl = dependency('libcurl',
+                       version : '>= 7.32.0',
+                       required : want_libcurl == 'yes')
+  if libcurl.found()
+    conf.set('HAVE_LIBCURL', 1)
+    m4_defines += ['-DHAVE_LIBCURL']
+  endif
+else
+  libcurl = declare_dependency('libcurl')
+endif
+
+want_libidn = get_option('libidn')
+if want_libidn != 'no'
+  libidn = dependency('libidn',
+                      required : want_libidn == 'yes')
+  if libidn.found()
+    conf.set('HAVE_LIBIDN', 1)
+    m4_defines += ['-DHAVE_LIBIDN']
+  endif
+else
+  libidn = declare_dependency('libidn')
+endif
+
+want_libiptc = get_option('libiptc')
+if want_libiptc != 'no'
+  libiptc = dependency('libiptc',
+                       required : want_libiptc == 'yes')
+  if libiptc.found()
+    conf.set('HAVE_LIBIPTC', 1)
+    m4_defines += ['-DHAVE_LIBIPTC']
+  endif
+else
+  libiptc = declare_dependency('iptc')
+endif
+
+want_qrencode = get_option('qrencode')
+if want_qrencode != 'no'
+  libqrencode = dependency('libqrencode',
+                           required : want_qrencode == 'yes')
+  if libqrencode.found()
+    conf.set('HAVE_QRENCODE', 1)
+  endif
+else
+  libqrencode = declare_dependency('libqrencode')
+endif
+
+want_gnutls = get_option('gnutls')
+if want_gnutls != 'no'
+  libgnutls = dependency('gnutls',
+                         version : '>= 3.1.4',
+                         required : want_gnutls == 'yes')
+  if libgnutls.found()
+    conf.set('HAVE_GNUTLS', 1)
+  endif
+else
+  libgnutls = declare_dependency('libgnutls')
+endif
+
+want_elfutils = get_option('elfutils')
+if want_elfutils != 'no'
+  libdw = dependency('libdw',
+                     required : want_elfutils == 'yes')
+  if libdw.found()
+    conf.set('HAVE_ELFUTILS', 1)
+  endif
+else
+  libdw = declare_dependency('libdw')
+endif
+
+want_zlib = get_option('zlib')
+if want_zlib != 'no'
+  libz = dependency('zlib',
+                    required : want_zlib == 'yes')
+  if libz.found()
+    conf.set('HAVE_ZLIB', 1)
+  endif
+else
+  libz = declare_dependency('zlib')
+endif
+
+want_bzip2 = get_option('bzip2')
+if want_bzip2 != 'no'
+  libbzip2 = dependency('bzip2',
```

Comment:
It's probably find to just always use `cc.find_library('bzip2')` since bzip2 doesn't require any special CFLAGS to compile against anyway.